recipes-kernel/linux/files/patches-6.1/0082-dts-iot2050-Support-IOT2050-SM-variant.patch

```diff
@@ -0,0 +1,400 @@
++	/delete-property/ pinctrl-8;
++	/delete-property/ pinctrl-9;
++	/delete-property/ pinctrl-10;
++	/delete-property/ pinctrl-11;
++	/delete-property/ pinctrl-12;
++	/delete-property/ pinctrl-13;
++	/delete-property/ pinctrl-14;
++	/delete-property/ pinctrl-15;
++	/delete-property/ pinctrl-16;
++	/delete-property/ pinctrl-17;
++	/delete-property/ pinctrl-18;
++	/delete-property/ pinctrl-19;
++	/delete-property/ pinctrl-20;
++	/delete-property/ pinctrl-21;
++	/delete-property/ pinctrl-22;
++	/delete-property/ pinctrl-23;
++	/delete-property/ pinctrl-24;
++
++	main_pcie_enable_pins_default: main-pcie-enable-default-pins {
++		pinctrl-single,pins = <
++			AM65X_IOPAD(0x01d8, PIN_OUTPUT, 7)  /* (AH12) GPIO1_22 */
++		>;
++	};
++
++	main_spi0_pins: main-spi0-default-pins  {
++		pinctrl-single,pins = <
++			AM65X_IOPAD(0x01c4, PIN_INPUT, 0) /* (AH13) SPI0_CLK */
++			AM65X_IOPAD(0x01c8, PIN_INPUT, 0) /* (AE13) SPI0_D0 */
++			AM65X_IOPAD(0x01cc, PIN_INPUT, 0) /* (AD13) SPI0_D1 */
++			AM65X_IOPAD(0x01bc, PIN_OUTPUT, 0) /* (AG13) SPI0_CS0 */
++		>;
++	};
++};
++
++&main_pmx1 {
++	asic_spi_mux_ctrl_pin: asic-spi-mux-ctrl-default-pins {
++		pinctrl-single,pins = <
++			AM65X_IOPAD(0x0010, PIN_OUTPUT, 7)  /* (D21) GPIO1_86 */
++		>;
++	};
++};
++
++&wkup_pmx0 {
++	/delete-property/ pinctrl-names;
++	/delete-property/ pinctrl-0;
++	/delete-property/ pinctrl-1;
++	/delete-property/ pinctrl-2;
++	/delete-property/ pinctrl-3;
++	/delete-property/ pinctrl-4;
++	/delete-property/ pinctrl-5;
++	/delete-property/ pinctrl-6;
++	/delete-property/ pinctrl-7;
++	/delete-property/ pinctrl-8;
++	/delete-property/ pinctrl-9;
++	/delete-property/ pinctrl-10;
++	/delete-property/ pinctrl-11;
++	/delete-property/ pinctrl-12;
++	/delete-property/ pinctrl-13;
++	/delete-property/ pinctrl-14;
++	/delete-property/ pinctrl-15;
++	/delete-property/ pinctrl-16;
++	/delete-property/ pinctrl-17;
++	/delete-property/ pinctrl-18;
++	/delete-property/ pinctrl-19;
++	/delete-property/ pinctrl-20;
++	/delete-property/ pinctrl-21;
++	/delete-property/ pinctrl-22;
++	/delete-property/ pinctrl-23;
++	/delete-property/ pinctrl-24;
++	/delete-property/ pinctrl-25;
++	/delete-property/ pinctrl-26;
++	/delete-property/ pinctrl-27;
++	/delete-property/ pinctrl-28;
++	/delete-property/ pinctrl-29;
++	/delete-property/ pinctrl-30;
++	/delete-property/ pinctrl-31;
++	/delete-property/ pinctrl-32;
++	/delete-property/ pinctrl-33;
++	/delete-property/ pinctrl-34;
++	/delete-property/ pinctrl-35;
++	/delete-property/ pinctrl-36;
++	/delete-property/ pinctrl-37;
++	/delete-property/ pinctrl-38;
++	/delete-property/ pinctrl-39;
++	/delete-property/ pinctrl-40;
++	/delete-property/ pinctrl-41;
++	/delete-property/ pinctrl-42;
++	/delete-property/ pinctrl-43;
++	/delete-property/ pinctrl-44;
++	/delete-property/ pinctrl-45;
++	/delete-property/ pinctrl-46;
++	/delete-property/ pinctrl-47;
++	/delete-property/ pinctrl-48;
++	/delete-property/ pinctrl-49;
++	/delete-property/ pinctrl-50;
```

Comment:
Quite a lot to delete, not only here. It may be better to factor out some iot2050-arduino.dtsi and include that only from the devices that support it.

Reply:
If factoring out. I find it's hard to deal with `arduino_i2c_aio_switch_pins_default` and `arduino_io_oe_pins_default`... Any ideas?